backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/deliveryplanning/DeliveryPlanningRepository.java

```diff
@@ -0,0 +1,129 @@
+		deliveryPlanningRecord.setM_ReceiptSchedule_ID(ReceiptScheduleId.toRepoId(request.getReceiptScheduleId()));
+		deliveryPlanningRecord.setM_ShipmentSchedule_ID(ShipmentScheduleId.toRepoId(request.getShipmentScheduleId()));
+		deliveryPlanningRecord.setC_Order_ID(OrderId.toRepoId(request.getOrderId()));
+		deliveryPlanningRecord.setC_OrderLine_ID(OrderLineId.toRepoId(request.getOrderLineId()));
+		deliveryPlanningRecord.setM_Product_ID(ProductId.toRepoId(request.getProductId()));
+		deliveryPlanningRecord.setM_Warehouse_ID(WarehouseId.toRepoId(request.getWarehouseId()));
+		deliveryPlanningRecord.setM_ShipperTransportation_ID(ShipperTransportationId.toRepoId(request.getShipperTransportationId()));
+		deliveryPlanningRecord.setC_BPartner_ID(BPartnerId.toRepoId(request.getPartnerId()));
+		deliveryPlanningRecord.setC_BPartner_Location_ID(BPartnerLocationId.toRepoId(request.getBPartnerLocationId()));
+		deliveryPlanningRecord.setC_Incoterms_ID(IncotermsId.toRepoId(request.getIncotermsId()));
+		deliveryPlanningRecord.setM_SectionCode_ID(SectionCodeId.toRepoId(request.getSectionCodeId()));
+
+		final LocalDateAndOrgId plannedDeliveryDate = request.getPlannedDeliveryDate();
+		final LocalDateAndOrgId actualDeliveryDate = request.getActualDeliveryDate();
+		final LocalDateAndOrgId plannedLoadingDate = request.getPlannedLoadingDate();
+		final LocalDateAndOrgId actualLoadingDate = request.getActualLoadingDate();
+
+		deliveryPlanningRecord.setPlannedDeliveryDate(plannedDeliveryDate == null ? null : plannedDeliveryDate.toTimestamp(orgDAO::getTimeZone));
+		deliveryPlanningRecord.setActualDeliveryDate(actualDeliveryDate == null ? null : actualDeliveryDate.toTimestamp(orgDAO::getTimeZone));
+		deliveryPlanningRecord.setPlannedLoadingDate(plannedLoadingDate == null ? null : plannedLoadingDate.toTimestamp(orgDAO::getTimeZone));
+		deliveryPlanningRecord.setActualLoadingDate(actualLoadingDate == null ? null : actualLoadingDate.toTimestamp(orgDAO::getTimeZone));
+
+		final Quantity qtyOrdered = request.getQtyOredered();
+		final Quantity qtyTotalOpen = request.getQtyTotalOpen();
+		final Quantity actualDeliveredQty = request.getActualDeliveredQty();
+		final Quantity actualLoadQty = request.getActualLoadQty();
+
+		deliveryPlanningRecord.setC_UOM_ID(request.getUom().getC_UOM_ID());
+
+		deliveryPlanningRecord.setQtyOrdered(qtyOrdered.toBigDecimal());
+		deliveryPlanningRecord.setQtyTotalOpen(qtyTotalOpen.toBigDecimal());
+		deliveryPlanningRecord.setActualDeliveredQty( actualDeliveredQty.toBigDecimal());
+		deliveryPlanningRecord.setActualLoadQty(actualLoadQty.toBigDecimal());
+
+		deliveryPlanningRecord.setForwarder(request.getForwarder());
+		deliveryPlanningRecord.setWayBillNo(request.getWayBillNo());
+		deliveryPlanningRecord.setReleaseNo(request.getReleaseNo());
+		deliveryPlanningRecord.setTransportDetails(request.getTransportDetails());
+
+		deliveryPlanningRecord.setIsActive(request.isActive);
```

Comment:
why not using the method?

and mainly why are we creating inactive records? what's the requirement?

Reply:
Got rd of it. It was not needed